internal/pkg/agent/install/uninstall_test.go

```diff
@@ -226,3 +227,22 @@ func TestNotifyFleetAuditUnenroll(t *testing.T) {
 
 	})
 }
+
+type MockNotifyFleetAuditUninstall struct {
+	Called bool
+}
+
+func (m *MockNotifyFleetAuditUninstall) Call(ctx context.Context, log *logp.Logger, pt *progressbar.ProgressBar, cfg *configuration.Configuration, ai fleetapi.AgentInfo) {
+	m.Called = true
+}
+
+func TestSkipFleetAuditUnenroll(t *testing.T) {
+	log := &logp.Logger{}
+	pt := &progressbar.ProgressBar{}
+	cfg := &configuration.Configuration{}
+	var agentID agentInfo = "testID"
+
+	mockNotify := &MockNotifyFleetAuditUninstall{}
+	notifyFleetIfNeeded(context.Background(), log, pt, cfg, agentID, true, false, true, notifyFleetAuditUninstall)
```

Comment:
let's also add checks with this matrix in mind, it should increase coverage a bit

expected run | notifyFleet | localFleet | skipFleetAudit
---|----|----|----
false | true | true | true
false | true | true | false
false | true | false | true
true | true | false | false
false | false | true | true
false | false | true | false
false | false | false | true
false | false | false | false

Reply:
other two files are more complex and not tested using Unit Tests so it may be we'll get sonarcube complains even after that but don't worry we'll handle that. let's just add this one table test

Reply:
you can get inspiration [here](https://github.com/elastic/elastic-agent/blob/c4835ca9d8f44fa93dbe2b19d7ec8110cecd1623/internal/pkg/agent/install/install_test.go#L69) when it comes to test form